transport/src/main/java/io/netty/channel/nio/NioIoHandler.java

```diff
@@ -653,43 +653,20 @@ private void select(IoExecutionContext runner, boolean oldWakenUp) throws IOExce
                     // - a scheduled task is ready for processing
                     break;
                 }
-                if (Thread.interrupted()) {
-                    // Thread was interrupted so reset selected keys and break so we not run into a busy loop.
-                    // As this is most likely a bug in the handler of the user or it's client library we will
-                    // also log it.
-                    //
-                    // See https://github.com/netty/netty/issues/2426
-                    if (logger.isDebugEnabled()) {
-                        logger.debug("Selector.select() returned prematurely because " +
-                                "Thread.currentThread().interrupt() was called. Use " +
-                                "NioHandler.shutdownGracefully() to shutdown the NioHandler.");
-                    }
-                    selectCnt = 1;
+                if (unexpectedSelectorWakeup(selectCnt)) {
                     break;
```

Comment:
We don't need to preserve the `selectCnt = 1;` line?

Likewise for the `} else if (selectReturnPrematurelyManyTimes(selectCnt)) {` branch further down.